core/src/main/java/org/infinispan/cache/impl/SimpleLocalCacheImpl.java

```diff
@@ -0,0 +1,1608 @@
+import org.infinispan.metadata.EmbeddedMetadata;
+import org.infinispan.metadata.Metadata;
+import org.infinispan.notifications.cachelistener.CacheNotifier;
+import org.infinispan.notifications.cachelistener.annotation.*;
+import org.infinispan.notifications.cachelistener.filter.CacheEventConverter;
+import org.infinispan.notifications.cachelistener.filter.CacheEventFilter;
+import org.infinispan.partitionhandling.AvailabilityMode;
+import org.infinispan.remoting.rpc.RpcManager;
+import org.infinispan.security.AuthorizationManager;
+import org.infinispan.stats.Stats;
+import org.infinispan.util.TimeService;
+import org.infinispan.util.concurrent.locks.LockManager;
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+import javax.transaction.TransactionManager;
+import javax.transaction.xa.XAResource;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+/**
+ * Simple local cache without interceptor stack.
+ * The cache still implements {@link AdvancedCache} since it is too troublesome to omit that.
+ *
+ * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+ */
+@MBean(objectName = CacheImpl.OBJECT_NAME, description = "Component that represents a simplified cache instance.")
+public class SimpleLocalCacheImpl<K, V> implements AdvancedCache<K, V> {
+	private final static Log log = LogFactory.getLog(SimpleLocalCacheImpl.class);
+	private final static Class<? extends Annotation>[] FIRED_EVENTS = new Class[] {
+			CacheEntryCreated.class, CacheEntryRemoved.class, CacheEntryVisited.class,
+	      CacheEntryModified.class, CacheEntriesEvicted.class, CacheEntryInvalidated.class };
+	private final String name;
+	private ComponentRegistry componentRegistry;
+	private Configuration configuration;
+	private EmbeddedCacheManager cacheManager;
+	private DataContainer<K, V> dataContainer;
+	private CacheNotifier<K, V> cacheNotifier;
+	private TimeService timeService;
+	private InternalEntryFactory entryFactory;
+
+	private Metadata defaultMetadata;
+	private Equivalence<Object> keyEquivalence;
+	private Equivalence<Object> valueEquivalence;
+	private Equivalence<Entry<Object, Object>> entryEquivalence;
+
+	private boolean hasListeners = false;
+
+	public SimpleLocalCacheImpl(String cacheName) {
+		this.name = cacheName;
+	}
+
+	@Inject
+	public void injectDependencies(
+			ComponentRegistry componentRegistry,
+			Configuration configuration,
+			EmbeddedCacheManager cacheManager,
+			DataContainer dataContainer,
+			CacheNotifier cacheNotifier,
+			TimeService timeService,
+			InternalEntryFactory entryFactory
+	) {
```

Comment:
We don't usually put the closing bracket on its own line (or the first parameter on a new line).